### PR TITLE
MultiCloud style

### PR DESCRIPTION
**Pull request by kferrone**
## **User description**
Although the branch is for azure. This is for gcp as well. This updates the setup action to properly setup the cli and login to the correct cloud.

___

## **Type**
enhancement, documentation

___

## **Description**
- Added support for AWS, GCP, and Azure in the setup action, including CLI installations and authentication steps.
- Expanded documentation to include detailed input variables, usage examples, and third-party actions used.
- Introduced Azure specific notes in the Terraform module README regarding the `RESOURCE_GROUP` environment variable.
- Enhanced the Terraform module action to support backend configurations for AWS, GCP, and Azure, including error handling for Azure setups.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Expanded Setup Action Documentation for AWS, GCP, and Azure</code></dd></summary>
<hr>
      
setup/README.md

<li>Expanded the setup action description to include support for AWS, GCP, <br>and Azure.<br> <li> Added a detailed inputs section for configuring the action.<br> <li> Provided usage examples for AWS, and a combined example for GCP or <br>Azure.<br> <li> Listed third-party actions used for AWS, GCP, and Azure setups.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/actions/pull/8/files#diff-df84bda362aea828518df16b8dcb81dccf27da8781f47a5aaf958f4050eec41f">+47/-13</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Added Azure Specific Notes to Terraform Module README</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
terraform-module/README.md

<li>Added notes for Azure users regarding the <code>RESOURCE_GROUP</code> environment <br>variable requirement.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/actions/pull/8/files#diff-6f28b477cc635af5d968dbc4dc576052372a08f8cba56b4e60baa3b69f0df930">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>action.yml</strong><dd><code>Support for AWS, GCP, and Azure in Setup Action</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
setup/action.yml

<li>Updated input descriptions and changed <code>gcp_credentials</code> to <code>credentials</code> <br>for clarity.<br> <li> Simplified installation steps and added cloud environment <br>configuration.<br> <li> Added AWS, GCP, and Azure specific configuration steps.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/actions/pull/8/files#diff-0a9aab6a50f3bbdec6e276c293c50c88f6fe9090b94442ca288b8d983bd19dc7">+62/-35</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>action.yml</strong><dd><code>Minor Adjustment to Terraform Exec Action Default Config Path</code></dd></summary>
<hr>
      
terraform-exec/action.yml

- Minor change to the default value of the `config` input.

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/actions/pull/8/files#diff-1253064d6a697ac79972242619fb2858bc6f1953b0a7d9033ab78877332ca875">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>action.yml</strong><dd><code>Enhanced Terraform Module Action for Multi-Cloud Support</code>&nbsp; </dd></summary>
<hr>
      
terraform-module/action.yml

<li>Added backend configuration for AWS, GCP, and Azure.<br> <li> Introduced error handling for missing <code>RESOURCE_GROUP</code> in Azure setups.<br> <li> Added caching support detection via Terraform lock file.<br> <li> Exported useful environment variables for subsequent steps.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/actions/pull/8/files#diff-0981b6ec692b63a24bfc693ec2240bb5bc682f5c3a96211b5eb1f5e79700013e">+25/-2</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table><hr>

<details> <summary><strong>✨ Usage guide:</strong></summary><hr> 

**Overview:**
The `describe` tool scans the PR code changes, and generates a description for the PR - title, type, summary, walkthrough and labels. The tool can be triggered [automatically](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) every time a new PR is opened, or can be invoked manually by commenting on a PR.

When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L46) related to the describe tool (`pr_description` section), use the following template:
```
/describe --pr_description.some_config1=... --pr_description.some_config2=...
```
With a [configuration file](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#working-with-github-app), use the following template:
```
[pr_description]
some_config1=...
some_config2=...
```

<table><tr><td><details> <summary><strong> Enabling\disabling automation </strong></summary><hr>

- When you first install the app, the [default mode](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) for the describe tool is:
```
pr_commands = ["/describe --pr_description.add_original_user_description=true" 
                         "--pr_description.keep_original_user_title=true", ...]
```
meaning the `describe` tool will run automatically on every PR, will keep the original title, and will add the original user description above the generated description. 

- Markers are an alternative way to control the generated description, to give maximal control to the user. If you set:
```
pr_commands = ["/describe --pr_description.use_description_markers=true", ...]
```
the tool will replace every marker of the form `pr_agent:marker_name` in the PR description with the relevant content, where `marker_name` is one of the following:
  - `type`: the PR type.
  - `summary`: the PR summary.
  - `walkthrough`: the PR walkthrough.

Note that when markers are enabled, if the original PR description does not contain any markers, the tool will not alter the description at all.
        

</details></td></tr>

<tr><td><details> <summary><strong> Custom labels </strong></summary><hr>

The default labels of the `describe` tool are quite generic: [`Bug fix`, `Tests`, `Enhancement`, `Documentation`, `Other`].

If you specify [custom labels](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md#handle-custom-labels-from-the-repos-labels-page-gem) in the repo's labels page or via configuration file, you can get tailored labels for your use cases.
Examples for custom labels:
- `Main topic:performance` - pr_agent:The main topic of this PR is performance
- `New endpoint` - pr_agent:A new endpoint was added in this PR
- `SQL query` - pr_agent:A new SQL query was added in this PR
- `Dockerfile changes` - pr_agent:The PR contains changes in the Dockerfile
- ...

The list above is eclectic, and aims to give an idea of different possibilities. Define custom labels that are relevant for your repo and use cases.
Note that Labels are not mutually exclusive, so you can add multiple label categories.
Make sure to provide proper title, and a detailed and well-phrased description for each label, so the tool will know when to suggest it.        

</details></td></tr>

<tr><td><details> <summary><strong> Inline File Walkthrough 💎</strong></summary><hr>

For enhanced user experience, the `describe` tool can add file summaries directly to the "Files changed" tab in the PR page.
This will enable you to quickly understand the changes in each file, while reviewing the code changes (diffs).

To enable inline file summary, set `pr_description.inline_file_summary` in the configuration file, possible values are:
- `'table'`: File changes walkthrough table will be displayed on the top of the "Files changed" tab, in addition to the "Conversation" tab.
- `true`: A collapsable file comment with changes title and a changes summary for each file in the PR.
- `false` (default): File changes walkthrough will be added only to the "Conversation" tab.
<tr><td><details> <summary><strong> Utilizing extra instructions</strong></summary><hr>

The `describe` tool can be configured with extra instructions, to guide the model to a feedback tailored to the needs of your project.

Be specific, clear, and concise in the instructions. With extra instructions, you are the prompter. Notice that the general structure of the description is fixed, and cannot be changed. Extra instructions can change the content or style of each sub-section of the PR description.

Examples for extra instructions:
```
[pr_description] 
extra_instructions="""
- The PR title should be in the format: '<PR type>: <title>'
- The title should be short and concise (up to 10 words)
- ...
"""
```
Use triple quotes to write multi-line instructions. Use bullet points to make the instructions more readable.

</details></td></tr>

<tr><td><details> <summary><strong> More PR-Agent commands</strong></summary><hr> 

> To invoke the PR-Agent, add a comment using one of the following commands:  
> - **/review**: Request a review of your Pull Request.   
> - **/describe**: Update the PR title and description based on the contents of the PR.   
> - **/improve [--extended]**: Suggest code improvements. Extended mode provides a higher quality feedback.   
> - **/ask \<QUESTION\>**: Ask a question about the PR.   
> - **/update_changelog**: Update the changelog based on the PR's contents.   
> - **/add_docs** 💎: Generate docstring for new components introduced in the PR.   
> - **/generate_labels** 💎: Generate labels for the PR based on the PR's contents.   
> - **/analyze** 💎: Automatically analyzes the PR, and presents changes walkthrough for each component.   

>See the [tools guide](https://github.com/Codium-ai/pr-agent/blob/main/docs/TOOLS_GUIDE.md) for more details.
>To list the possible configuration parameters, add a **/config** comment.   
 

</details></td></tr>

</table>

See the [describe usage](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md) page for a comprehensive guide on using this tool.

</details>

### CODE REVIEW

File: setup/action.yml [hunk 1]
```diff
@@ -11,9 +11,9 @@ inputs:
   account-id: 
     required: false
     description: Overide the cloud account id from the default. Required when on gcp where this would be the project name. 
-  gcp_credentials: 
+  credentials: 
     required: false
-    description: GCP credentials json. Required when on gcp.
```
Comment (corymay-duplocloud) [inline setup/action.yml:16]: Caps on GCP for consistency?

Comment (kferrone) [reply setup/action.yml:16]: This is what I changed it from. See the new changes. It doesn't even have this anymore.

Comment (corymay-duplocloud) [reply setup/action.yml:16]: face palm ofc

**Review by corymay-duplocloud — APPROVED** [review]

**Review by corymay-duplocloud — APPROVED** [review]

**Review by jmaggio — APPROVED** [review]
looks good to me, lets just make sure to verify this branch on a real pipeline first for each target cloud

**Pull request merged**